### DISCUSSION TITLE
id cliente da elenco fatture

### DISCUSSION THREAD

**Discussion by PaoloPLG** [fattureincloud/api#313] [Q&A] [answered]
Buongiorno a tutti,

connettendomi tramite le API V2 riesco a scaricare l'elenco delle fatture ma tra i dati non è presente l'id cliente.
Per ottenere l'id cliente l'unico modo è quello di lanciare una richiesta di dettaglio fattura per ogni singola fattura?

Grazie mille,
Paolo.

**Comment by emajo** [comment]
Ciao @PaoloPLG, l'id del cliente viene restituito anche richiedendo l'elenco delle fatture, il campo in questione è `entity.id`, se questo campo è valorizzato a null significa che non è stato passato in creazione del documento. 
Il comportamento riscontrato risulta differente?

Grazie,
Emanuele.

**Reply by PaoloPLG** [reply]
Ciao Emanuele,

innanzitutto grazie per la risposta.

Nei dati restituiti dalle API ho un campo "entity" (non "entity.id") che è sempre vuoto per tutte le fatture.

Quando le fatture vengono create (ad oggi sempre manualmente) viene sempre assegnato un cliente quindi non capisco bene cosa intendi quando dici "se questo campo è valorizzato a null significa che non è stato passato in creazione del documento."

Grazie mille,
Paolo.

**Reply by emajo** [reply]
Ciao @PaoloPLG, se una fattura viene creata senza collegare un cliente presente in anagrafica, il risultato ritornato nella risposta della chiamata di elenco fatture sarà simile alla seguente: 
![fattura no id](https://user-images.githubusercontent.com/48432054/234585995-facda468-8452-494c-a1d7-577729d0ed14.png)
se invece viene collegato un cliente tramite l'apposito tasto (se l'operazione viene fatta manualmente) o passando l'id del cliente in creazione documento (tramite API) allora il risultato sarà simile al seguente:
![fattura id](https://user-images.githubusercontent.com/48432054/234587648-0d43d66d-4ef1-4210-abe8-2541e65af147.png)
Come può vedere nel secondo caso è stato restituito anche il campo id all'interno dell'oggetto entity, solitamente il campo entity non viene mai restituito vuoto.
Che strumento sta usando per fare queste chiamate? In caso le chiedo di condividerci un esempio di risposta per indagare più a fondo.

Grazie,
Emanuele.

**Reply by PaoloPLG** [reply, marked as answer]
Ciao Emanuele,

ho trovato il problema ed era un mio errore.
Avevo valorizzato il campo "fields" della funzione "listIssuedDocuments" in modo errato ed il campo "entity" veniva sempre mostrato vuoto.

Grazie per l'assistenza,
Paolo.
